debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -165,7 +168,7 @@ public SnapshotResult<O> doExecute(ChangeEventSourceContext context, O previousO
             dispatcher.alwaysDispatchHeartbeatEvent(ctx.partition, ctx.offset);
             return SnapshotResult.completed(ctx.offset);
         }
-        catch (final Exception e) {
+        catch (final Throwable e) {
```

Comment:
Why `Throwable`?

Reply:
To capture the `AssertionError` that extends Error > Throwable

For example
![image](https://github.com/debezium/debezium/assets/2572323/5dfb7506-9c92-4418-8ad5-b233dd910630)

this happens in the postSnapshot method and in case of assertion failure it was hidden.

Reply:
In that case I'd prefer to have it as an explicit case. not generic throwable as most of the errors like OOME is not something you should handle at all.